app/src/main/java/com/edu/sun/oereminder/utils/JsonExtension.kt

```diff
@@ -0,0 +1,86 @@
+package com.edu.sun.oereminder.utils
+
+import com.edu.sun.oereminder.data.model.Account
+import com.edu.sun.oereminder.data.model.Member
+import com.edu.sun.oereminder.data.model.Message
+import com.edu.sun.oereminder.data.model.Room
+import com.edu.sun.oereminder.utils.JsonKey.ACCOUNT_ID
+import com.edu.sun.oereminder.utils.JsonKey.AVATAR_IMAGE_URL
+import com.edu.sun.oereminder.utils.JsonKey.BODY
+import com.edu.sun.oereminder.utils.JsonKey.FILE_NUM
+import com.edu.sun.oereminder.utils.JsonKey.ICON_PATH
+import com.edu.sun.oereminder.utils.JsonKey.LAST_UPDATE_TIME
+import com.edu.sun.oereminder.utils.JsonKey.LOGIN_MAIL
+import com.edu.sun.oereminder.utils.JsonKey.MENTION_NUM
+import com.edu.sun.oereminder.utils.JsonKey.MESSAGE_ID
+import com.edu.sun.oereminder.utils.JsonKey.MESSAGE_NUM
+import com.edu.sun.oereminder.utils.JsonKey.NAME
+import com.edu.sun.oereminder.utils.JsonKey.ROLE
+import com.edu.sun.oereminder.utils.JsonKey.ROOM_ID
+import com.edu.sun.oereminder.utils.JsonKey.SEND_TIME
+import com.edu.sun.oereminder.utils.JsonKey.STICKY
+import com.edu.sun.oereminder.utils.JsonKey.TASK_NUM
+import com.edu.sun.oereminder.utils.JsonKey.TYPE
+import com.edu.sun.oereminder.utils.JsonKey.UNREAD_NUM
+import com.edu.sun.oereminder.utils.JsonKey.UPDATE_TIME
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+@Throws(JSONException::class)
+inline fun <reified T> String.toListData() = with(JSONArray(this)) {
+    List(length()) {
+        with(getJSONObject(it)) {
+            when (T::class) {
+                Message::class -> toMessage() as T
+                Member::class -> toMember() as T
+                Room::class -> toRoom() as T
+                Account::class -> toAccount() as T
+                else -> throw JSONException("Type not supported")
+            }
+        }
+    }
+}
+
+@Throws(JSONException::class)
+fun JSONObject.toAccount() = Account(
+    getInt(ACCOUNT_ID),
+    getString(AVATAR_IMAGE_URL),
+    getString(LOGIN_MAIL),
+    getString(NAME)
+)
```

Comment:
những cái này nên tạo constructor của từng class sẽ tốt hơn

Reply:
Em Throw JSONException ở đây để không phải Throw Exception từ constructor anh ạ.

Reply:
Thêm 1 constructor trong class Account 
```suggestion
Account(jsonObject)
```